### PR TITLE
Numbers that starts with . should be allowed

### LINKED ISSUE #225: [BUG]  Numbers that starts with . should be allowed
**Describe the bug**
Entering a value the starts with a decimal/dot causes an error: "Error, invalid characters".

**To Reproduce**
1. Go to "Length"
2. Click on "Meters"
3. Enter ".5"
5. See error

**Expected behavior**
A leading zero should be implied when a leading decimal/dot is entered. eg ".5" should be read as "0.5". This seems to apply to all units.

**Screenshots / gif**
N/A

**Device**
- Device type: Android
- OS version: Android 10
- Device type: Moto G Power

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by HarshMakwana27**
Issue fixed by changes in onChanged and validator.

Fix #225

### CODE REVIEW

File: android/gradle/wrapper/gradle-wrapper.properties
Comment (ferraridamiano) [inline android/gradle/wrapper/gradle-wrapper.properties:1]: Could you please revert the changes on this file? Gradle upgrade should not be included in this PR

File: pubspec.yaml
Comment (ferraridamiano) [inline pubspec.yaml:1]: Could you please revert the changes on this file?

If you do as you suggest we ends up including all the svg files + their compiled version while we would like to keep just the optimized one

**Review by ferraridamiano — CHANGES_REQUESTED** [review]
Looks good! Just leaved few comments

File: pubspec.yaml
Comment (HarshMakwana27) [reply pubspec.yaml:1]: fixed

**Review by ferraridamiano — APPROVED** [review]

**Comment by ferraridamiano** [comment]
Thank you @HarshMakwana27
There are a lot of issues like this tha could be fixed if you want :)

**Pull request merged**